### ISSUE TITLE
Default Language to System Default

### ISSUE THREAD

**Issue by maymage** [hugolabe/Wike#74]
Hi all, 

I have just installed the latest version of Wike `Version 1.7.2 (2d324b86)` from flathub and to try it out
searched for `caesar`. Wike correctly showed me the article on `Gaius Julius Caesar`. However, 

* the article was in English,  while 
* my system default is set to German. 
* Checking in `Wike -> Settings -> Languages` German was the only selected language and
* only after selecting German from the wheel in the search bar, results were persistently shown in German. 

I honestly think, that the search results should have been in German from the very start.  So here my feature request

**Default to the system language at first use**

---

Once again thanks for bringing wikipedia on GNOME - I use it daily!

**Comment by hugolabe** [comment]
Yes, that is the normal behavior of the application.

Setting the system language as default I think is not a good idea. As I [said](https://github.com/hugolabe/Wike/issues/35#issuecomment-910453245) in another issue, many of the local wikipedias have very little content and their users will probably use English or other languages. I think it is better that the user chooses which language he wants to use for searches.

**Comment by maymage** [comment]
I see - here two suggestions

* default to system language, if that language has more than a certain amount of articles - say 10^6
* always default to the system language and 
* show the English Wikipedia article in case there is no article to be found in the system language

**Comment by hugolabe** [comment]
> * default to system language, if that language has more than a certain amount of articles - say 10^6

That would make it necessary to keep track of the number of articles that each wikipedia has. And there are almost 300!

> * show the English Wikipedia article in case there is no article to be found in the system language

I believe that the behavior of an application should be predictable and consistent. It doesn't seem right to me that searches in one language return a result in a different one.

**Comment by maymage** [comment]
> That would make it necessary to keep track of the number of articles that each wikipedia has. And there are almost 300!

Fun fact: there's a wikipedia [article](https://meta.wikimedia.org/wiki/List_of_Wikipedias) ranking wikipedia articles per language with a list of those over 10^6 containing currently 18 languages with I'd say more than 500*10^6 speakers. 

---

> I believe that the behavior of an application should be predictable and consistent. It doesn't seem right to me that searches in one language return a result in a different one.

I see and so I'm just brainstorming here for predicability.
 
What about supporting, e.g. n=#(m_1,...,m_n) languages such that for a search query

* `no article was found` results in case there is none to be found in any of the languages m_1,...,m_n
* the article in the first language is shown, if existent, i.e. the smallest p=1,...,m for which we find a corresponding article in language m_p. 

The default would be n=2 with 

* m_1 = system default language
* m_2 = English

 --- 

Personally I check wikipedias in the following order

* German = mother tongue
* Italian = second language, that I know best
* English as the next best thing. 

---

Also the proposed default 

* first system language
* English as a second choice

perhaps- I can't prove it -  reflects a substantial real world usage: check in your own language first and then in English, because English has most Wikipedia articles and is known most known secondary language on this planet.

**Comment by hugolabe** [comment]
I think this is overcomplicating searches. In addition, each language is a different Wikipedia, so these would be separate searches and could cause performance issues.

Although it’s not the same, an option that could help those who use different languages would be the use of search prefixes, as they already exist in desktop searches.

This way you could default to the language you use the most and search in other languages by adding a prefix (eg -it to search in Italian).

**Comment by maymage** [comment]
> I think this is overcomplicating searches. In addition, each language is a different Wikipedia, so these would be separate searches and could cause performance issues.

Why would a second search in e.g. English, **only** if the first search in the default language, hamper performance?
Perhaps with an intermediate message: 

>`A corresponding article was not found. Would you like to search in the [second language, default English] instead?`

I really think, this could be a feature, not a bug. 

Of course this should be configurable -  being able to modify first and second language and to disable the second one altogether. 


---

> Although it’s not the same, an option that could help those who use different languages would be the use of search prefixes, as they already exist in desktop searches.

Oh, I didn't know, that was possible..

---

Please, just close this issue, if you think the proposal is not feasible or is otherwise unsuitable. I just made a suggestion here, 
there's no bug. 

Best

**Comment by hugolabe** [comment]
Yes, I understand that you propose it as an improvement and I thank you for the suggestion. But in my opinion the search should be kept simple and predictable.
